pennylane_lightning/src/rework/Util.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file
+ * Contains uncategorised utility functions.
+ */
+#pragma once
+
+#include <assert.h>
+
+namespace Pennylane {
+
+    /**
+     * Calculates 2^n for some integer n > 0 using bitshifts.
+     * 
+     * @param n the exponent
+     * @return value of 2^n
+     */
+    inline size_t exp2(const unsigned int& n) {
+        return (size_t)1 << n;
+    }
+
+    /**
+     * Calculates the decimal value for a qubit, assuming a big-endian convention.
```

Comment:
I had given this some thought in the past, but could never arrive at a more satisfactory term. It's really just trying to compute something like 0100 -> 2^2 = 4, i.e. the decimal value you'd add if the bit was 1.

Reply:
Makes sense! How about `maxDecimalForQubit` or `basisStateToDecimal`? It seems that we need both `qubits` and `qubitIndex` to get the result in the big-endian convention (and for `assert(qubitIndex < qubits)`). Otherwise, if we had assumed a little-endian convention, we could go with `exp2(qubitIndex)`. Also, `qubitIndex` determines the index in the `qubit` long system where we have a `1` (on the other registers we have `0`). So basically `qubitIndex` specifies a computational basis state that we are converting to the decimal representation.

Reply:
> Otherwise, if we had assumed a little-endian convention, we could go with exp2(qubitIndex)

I assumed that wasn't an option, short of having different conventions between the Python side of things (which I assumed interpret qubits in a big-endian fashion) and the C++ backend.

Reply:
> I assumed that wasn't an option, short of having different conventions between the Python side of things (which I assumed interpret qubits in a big-endian fashion) and the C++ backend.

Absolutely, having it big-endian is a great approach.

> Otherwise, if we had assumed a little-endian convention, we could go with exp2(qubitIndex)

This comment was just further considering what is being done within the function to get closer to a descriptive name, the current implementation is great as is.